### PR TITLE
`log` attribute macros for methods tracing.

### PR DESCRIPTION
**Pull request by humb1t**
Signed-off-by: Nikita Puzankov <humb1t@yandex.ru>

### CODE REVIEW

File: iroha_macro/iroha_derive/src/lib.rs [hunk 1]
```diff
@@ -2,7 +2,19 @@ extern crate proc_macro;
...
+    PatTuple, PatTupleStruct, PatType, Signature,
+};
+
+#[proc_macro_attribute]
+pub fn log(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    //if cfg!(debuginfo = "2") {
+    impl_log(item)
+    /*} else {
+        item
+    }*/
```
Comment (eadventurous) [inline iroha_macro/iroha_derive/src/lib.rs:16]: I think you forgot to remove commented code here.

File: iroha_macro/iroha_derive/tests/log.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#[cfg(test)]
+mod tests {
+
+    use iroha_derive::*;
+
+    #[log]
```
Comment (eadventurous) [inline iroha_macro/iroha_derive/tests/log.rs:6]: Seems like there is no straightforward way to test this macro as we need to check stdout

Comment (eadventurous) [reply iroha_macro/iroha_derive/tests/log.rs:6]: Maybe use https://crates.io/crates/gag . But it has a limitation that it runs only on unix systems.

Comment (humb1t) [reply iroha_macro/iroha_derive/tests/log.rs:6]: I think it's okay for now to only compile this and check in other tests

**Pull request merged**